--- pageredirect/syntax.php~	2007-01-24 12:36:50.000000000 +0200
+++ pageredirect/syntax.php	2008-01-25 11:36:14.597898649 +0200
@@ -36,7 +36,7 @@
 	function getSort() { return 1; }
 	
 	function connectTo($mode) { 
-		$this->Lexer->addSpecialPattern('~~REDIRECT>[a-zA-Z0-9_\-:]+~~', $mode, 'plugin_pageredirect');
+		$this->Lexer->addSpecialPattern('~~REDIRECT>.+~~', $mode, 'plugin_pageredirect');
 	}
 	
 	/** 
